pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -1752,8 +1760,12 @@ type ProtocolSpec struct {
 	Swift *SwiftSpec `json:"swift"`
 }
 
+// +kubebuilder:validation:Enum=s3;s3website;swift;swift_auth;admin;sts;iam;notifications
+type ObjectStoreAPI string
+
 // S3Spec represents Ceph Object Store specification for the S3 API
 type S3Spec struct {
+	// Deprecated: use protocol.enableAPIs instead.
```

Comment:
Is it ok to deprecate?

Reply:
I think swift support is marked experimental, so this can be deprecated without concern. However, I would think maybe we don't backport that to 1.15 if so. 

This would be a good discussion to include @kayrus and @jklippel in I think.